test/Interop/Cxx/templates/dependent-types.swift

```diff
@@ -85,4 +84,37 @@ DependentTypesTestSuite.test("Function template methods (static)") {
 }
 #endif // Windows
 
+DependentTypesTestSuite.test("Complex different dependent return type inferrd.") {
+  let m = complexDependantReturnTypeInferred(M<Int>(value: 42)) as! M<Int>
+  expectEqual(m.getValue(), 42)
+}
+
+// TODO: Currently still failing: Could not cast value of type '__C.__CxxTemplateInst1MIlE' to 'Swift.Int'
+// DependentTypesTestSuite.test("Complex different dependent argument and return type") {
+//   let m = complexDifferentDependentArgAndRet(M<Int>(value: 42), T: Int.self, U: Int.self) as! M<Int>
```

Comment:
This should return `as! Int`, I think.

Reply:
Unfortunately, getting the same error when casting to Int